src/ServerExtension.php

```diff
@@ -78,20 +76,24 @@ private static function reset(): void
         self::$registeredClients = [];
     }
 
-    private function takeScreenshots(string $type, string $test): void
+    public static function takeScreenshots(string $type, string $test): void
     {
-        if (!($_SERVER['PANTHER_ERROR_SCREENSHOT_DIR'] ?? false)) {
+        if (!self::$enabled || !($_SERVER['PANTHER_ERROR_SCREENSHOT_DIR'] ?? false)) {
             return;
         }
 
         foreach (self::$registeredClients as $i => $client) {
-            $client->takeScreenshot(sprintf('%s/%s_%s_%s-%d.png',
+            $screenshotPath = sprintf('%s/%s_%s_%s-%d.png',
                 $_SERVER['PANTHER_ERROR_SCREENSHOT_DIR'],
                 date('Y-m-d_H-i-s'),
                 $type,
                 strtr($test, ['\\' => '-', ':' => '_']),
                 $i
-            ));
+            );
+            $client->takeScreenshot($screenshotPath);
+            if ($_SERVER['PANTHER_ERROR_SCREENSHOT_ATTACH'] ?? false) {
```

Comment:
Is there a drawback to always call `printf`?

Reply:
PHPunit has `--disallow-test-output` option on the command line and `beStrictAboutOutputDuringTests="true"` setting in xml. Additionally, there's an `expectOutputString` method in phpunit (though I doubt someone would use it with panther).